intellij-plugin-structure/tests/src/test/kotlin/com/jetbrains/plugin/structure/intellij/PluginXmlValidationTest.kt

```diff
@@ -189,6 +194,134 @@ class PluginXmlValidationTest {
     assertEquals(emptyList<PluginProblem>(), pluginCreationSuccess.warnings.filterIsInstance<NoModuleDependencies>())
   }
 
+  @Test
+  fun `plugin with os and arch constraints dependencies`() {
+    val pluginCreationSuccess = buildCorrectPlugin {
+      dir("META-INF") {
+        file("plugin.xml") {
+          """
+            <idea-plugin>
+              $HEADER
+              <depends>com.intellij.modules.arch.x86_64</depends>
+              <depends>com.intellij.modules.os.macos</depends>
+            </idea-plugin>
+          """
+        }
+      }
+    }
+
+    assertEquals(emptyList<PluginProblem>(), pluginCreationSuccess.unacceptableWarnings)
+    assertEquals(emptyList<PluginProblem>(), pluginCreationSuccess.warnings)
+
+    assertEquals(CpuArch.X86_64, pluginCreationSuccess.plugin.archConstraints.single())
+    assertEquals(OS.MacOS, pluginCreationSuccess.plugin.osConstraints.single())
+  }
+
+  @Test
+  fun `plugin with os and arch optional constraints dependencies`() {
+    val pluginCreationSuccess = buildCorrectPlugin {
+      dir("META-INF") {
+        file("plugin.xml") {
+          """
+            <idea-plugin>
+              $HEADER
+              <depends optional="true" config-file="test.xml">com.intellij.modules.arch.x86_64</depends>
+              <depends optional="true" config-file="test2.xml">com.intellij.modules.os.macos</depends>
+            </idea-plugin>
+          """
+        }
+        file("test.xml") {
+          """
+            <idea-plugin>
+            </idea-plugin>
+          """.trimIndent()
+        }
+        file("test2.xml") {
+          """
+            <idea-plugin>
+            </idea-plugin>
+          """.trimIndent()
+        }
+      }
+    }
+
+    assertEquals(emptyList<PluginProblem>(), pluginCreationSuccess.unacceptableWarnings)
+    assertEquals(emptyList<PluginProblem>(), pluginCreationSuccess.warnings)
+
+    assertTrue(pluginCreationSuccess.plugin.archConstraints.isEmpty())
+    assertTrue(pluginCreationSuccess.plugin.osConstraints.isEmpty())
+  }
+
+  @Test
+  fun `plugin with os and arch v2 constraints dependencies`() {
+    val pluginCreationSuccess = buildCorrectPlugin {
+      dir("META-INF") {
+        file("plugin.xml") {
+          """
+            <idea-plugin>
+              $HEADER
+              
+              <dependencies>
+                <module name="com.intellij.modules.arch.x86_64" />
```

Comment:
these modules can't be referenced as v2 module dependencies, only as v2 plugin dependencies (they are plugin ids)